scripts/zones/Caedarva_Mire/npcs/qm8.lua

```diff
@@ -0,0 +1,35 @@
+-----------------------------------
+-- Area: Caedarva Mire
+-- NPC:  qm8
+-- Gives Lamian Fang Key
+-----------------------------------
+package.loaded["scripts/zones/Caedarva_Mire/TextIDs"] = nil
+-----------------------------------
+require("scripts/zones/Caedarva_Mire/TextIDs")
+
+function onTrade(player,npc,trade)
+end
+
+function onTrigger(player,npc)
```

Comment:
with scripts/globals/npc_util required,
```
function onTrigger(player,npc)
    if player:getVar("[TIMER]Lamian_Fang_Key") ~= VanadielDayOfTheYear() and npcUtil.giveItem(player, 2219) then
        player:setVar("[TIMER]Lamian_Fang_Key", VanadielDayOfTheYear()) -- Can obtain key once per vanadiel day
    else
        player:messageSpecial(NOTHING_OUT_OF_ORDINARY)
    end
end
```

Reply:
easy enough...

Reply:
ok so I had to change this script example given to me in order to get it working somewhat properly and while messing with this system, this is as close as I could get it to retail:
```
function onTrigger(player,npc)
    if player:getVar("[TIMER]Lamian_Fang_Key") ~= VanadielDayOfTheYear() then 
        if npcUtil.giveItem(player, 2219) then
            player:setVar("[TIMER]Lamian_Fang_Key", VanadielDayOfTheYear()) -- Can obtain key once per vanadiel day
        end
    else
        player:messageSpecial(NOTHING_OUT_OF_ORDINARY)
    end
end
```
But, it still isn't right as you still get the "Obtained:" message even with the Key already in your posession. The way I scripted it originally gives the proper retail responses.

Reply:
What's retail behavior when being given a rare item they already have?  I can update npcUtil.giveItem() to match.

Reply:
Awesome! Let me try it.

Reply:
It works... but... eh. I'd really like it to be retail accurate which is what i thought the goal of dsp was. the way i have it scripted originally is retail accurate. if you're interested, this is what the message from retail shows when you click on it while having a key already in your inventory:
![lamian_key](https://user-images.githubusercontent.com/30469395/40568353-c9f47114-603f-11e8-8a6e-9a36c2c75867.jpg)
get npc_utils to behave that way and i'll change it, but i don't see any reason to make it less like retail just to use the npc_util.

Reply:
[This](https://github.com/DarkstarProject/darkstar/pull/4904) should make npcUtil.giveItem behave like you want, though it only works when single item is being given (see discussion on #4889).

Reply:
it's not like how I want, it's like how dsp wants... we are on the same team.

Reply:
thank you! by the way... =)